slack_bolt/listener_matcher/builtins.py

```diff
@@ -235,28 +236,66 @@ def func(body: Dict[str, Any]) -> bool:
     raise BoltError(f"action ({constraints}: {type(constraints)}) must be any of str, Pattern, and dict")
 
 
-def _function_block_action(
-    callback_id: Union[str, Pattern],
+def function_block_action(
+    callback_id: str,
     constraints: Union[str, Pattern, Dict[str, Union[str, Pattern]]],
-    body: Dict[str, Any],
-) -> bool:
-    if is_function_interactivity(body) is False and is_block_actions(body) is False:
-        return False
+    asyncio: bool = False,
+    base_logger: Optional[Logger] = None,
+) -> Union[ListenerMatcher, "AsyncListenerMatcher"]:
+    def func(body: Dict[str, Any]) -> bool:
+        return (
+            _function_interactivity_match(callback_id, body)
+            and is_block_actions(body)
+            and _does_block_id_match(constraints, body)
+        )
 
-    if _matches(callback_id, body.get("function_data", {}).get("function", {}).get("callback_id", "")) is False:
-        return False
+    return build_listener_matcher(func, asyncio, base_logger)
 
-    return _does_block_id_match(constraints, body)
 
+def function_view(
```

Comment:
Thinking of refactoring this and the `view` function since the code is similar in order reduce code duplication 💡 
Maybe a factory method pattern could work, but it will definitely increase complexity